test/run

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/make -f
+
+# This is the entry point for the Cockpit integration test machinery.
+#
+# To run the tests manually and individually, you can do something
+# like this:
+#
+# $ export TEST_OS=rhel-8-0
+# $ ./test/run prepare
+# $ ./test/check-subscriptions -v -j1 -t
+
+ifeq ($(TEST_OS),)
+TEST_OS = rhel-8-0
+endif
+export TEST_OS
+VM_IMAGE=$(CURDIR)/test/images/$(TEST_OS).qcow2
+TAR=$(CURDIR)/cockpit/subscription-manager-cockpit.tar.gz
+
+TEST_NODE_MODULES = node_modules/chrome-remote-interface node_modules/sizzle
+
+check: prepare
+	test/check-subscriptions
+
+reset:
+	rm -f $(TAR) $(VM_IMAGE)
+
+prepare: reset $(VM_IMAGE) test/common $(TEST_NODE_MODULES)
+
+$(TAR): cockpit/node_modules
+	make -C cockpit dist-gzip
+
+$(VM_IMAGE): $(TAR) bots
+	rm -f $(VM_IMAGE)
+	bots/image-customize -v -u $(TAR):/var/tmp/ -s ./test/vm.install $(TEST_OS)
+
+# checkout Cockpit's bots/ directory for standard test VM images and API to launch them
+# must be from cockpit's master, as only that has current and existing images; but testvm.py API is stable
+bots:
+	git fetch --depth=1 https://github.com/cockpit-project/cockpit.git
+	git checkout --force FETCH_HEAD -- bots/
+	git reset bots
```

Comment:
> We also have Vagrant environment. Could it be possible to use existing Vagrant machine or could it be possible to create new testing machine using Vagrant?

I'll check that out.  The immediate goal here is to integrate the subscription-manager UI tests into the Cockpit CI machinery, because that would be relatively easy.  (One might say that it's done already. :-)

Does your Jenkins CI machinery use these Vagrant images?  If your are more comfortable with running Cockpit UI tests there, we can certainly work on that as well.

The Cockpit CI integration here has the downside that it only tests the new Cockpit code; it doesn't build and install a new rhsmd, for example.  I think that's a significant drawback, and if going with Jenkins would fix that, that would be a strong argument.

Reply:
> I'll check that out. The immediate goal here is to integrate the subscription-manager UI tests into the Cockpit CI machinery, because that would be relatively easy. (One might say that it's done already. :-)

I can see the reason, why you decided to use something else than Vagrant.
 
> Does your Jenkins CI machinery use these Vagrant images? If your are more comfortable with running Cockpit UI tests there, we can certainly work on that as well.

The definitions of Jenkins jobs are here: https://github.com/candlepin/candlepin-jobs/. Vagrant can be used for testing Jenkins jobs, but our Vagrant machines are used for Jenkins CI. Vagrant machines are used for testing mostly be developers.

> The Cockpit CI integration here has the downside that it only tests the new Cockpit code; it doesn't build and install a new rhsmd, for example. I think that's a significant drawback, and if going with Jenkins would fix that, that would be a strong argument.

Yeah, when Vagrant machine is used, then rhsmd and everything else is just as fresh as possible.

Reply:
I have added a commit here that installs rhsm into our VM image.